File: 1422-maximum-score-after-splitting-a-string/Yescafe.cc
class Solution {
public:
    int maxScore(string s) {
        int score = count(s.begin(), s.end(), '1');
        int ans = 0;
        int N = s.length();
        for (int i = 0; i < N - 1; ++i) {
            switch (s[i]) {
                case '0': ++score; break;
                case '1': --score; break;
            }
            ans = max(ans, score);
        }
    return ans;
    }
};

File: 1422-maximum-score-after-splitting-a-string/Yescafe.go
func maxScore(s string) int {
    ss := []byte(s)
    var max int
    score := bytes.Count(ss, []byte("1"))
    for _, c := range ss[:len(ss)-1] {
        switch c {
        case '1':
            score--
        case '0':
            score++
        }
        if score > max {
            max = score
        }
    }
    return max
}

